src/_types.ts

```diff
@@ -57,4 +71,14 @@ export interface Platform {
   toPlatformPointer<T>(value: Pointer<T> | null): PlatformPointer<T> | null;
 
   toPlatformString(value: string): PlatformString;
+
+  toPlatformStruct(data: TypedArray, _: unknown): Uint8Array;
+  toPlatformStruct<T extends AllocatableStruct>(
+    data: Pointer<T>,
+    dataType: AllocatableStructConstructor<T>,
+  ): Uint8Array;
+  toPlatformStruct<T extends AllocatableStruct>(
+    data: TypedArray | Pointer<T>,
+    dataType: AllocatableStructConstructor<T>,
+  ): Uint8Array;
```

Comment:
Suggestion: Let's just condense all these down to one:

```suggestion
toPlatformStruct<T extends AllocatableStruct>(
    data: TypedArray | Pointer<T>,
    dataType?: AllocatableStructConstructor<T>,
  ): Uint8Array;
```